File: backend/src/models/Patient.js
const { DataTypes } = require('sequelize');

module.exports = (sequelize) => {
  const Patient = sequelize.define('Patient', {
    id: {
      type: DataTypes.UUID,
      defaultValue: DataTypes.UUIDV4,
      primaryKey: true,
    },
    user_id: {
      type: DataTypes.UUID,
      allowNull: false,
      unique: true,
      references: {
        model: 'users',
        key: 'id',
      },
    },
    address: {
      type: DataTypes.TEXT,
      allowNull: true,
    },
    medical_history: {
      type: DataTypes.TEXT,
      allowNull: true,
    },
    emergency_contact: {
      type: DataTypes.STRING(100),
      allowNull: true,
    },
    dob: {
      type: DataTypes.DATEONLY,
      allowNull: true,
      validate: {
        isDate: { msg: 'Invalid date format' },
        isBefore: {
          args: new Date().toISOString().split('T')[0],
          msg: 'Date of birth must be in the past',
        },
      },
    },
  }, {
    tableName: 'patients',
    timestamps: true,
    underscored: true,
    indexes: [
      { unique: true, fields: ['user_id'] },
    ],
  });

  return Patient;
};

File: mobile/lib/models/patient.dart
import 'user.dart';

class Patient {
  final String id;
  final String userId;
  final String? address;
  final String? medicalHistory;
  final String? emergencyContact;
  final String? dob;
  final User? user;

  Patient({
    required this.id,
    required this.userId,
    this.address,
    this.medicalHistory,
    this.emergencyContact,
    this.dob,
    this.user,
  });

  factory Patient.fromJson(Map<String, dynamic> json) {
    return Patient(
      id: json['id'] ?? '',
      userId: json['user_id'] ?? '',
      address: json['address'],
      medicalHistory: json['medical_history'],
      emergencyContact: json['emergency_contact'],
      dob: json['dob'],
      user: json['user'] != null ? User.fromJson(json['user']) : null,
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'user_id': userId,
      'address': address,
      'medical_history': medicalHistory,
      'emergency_contact': emergencyContact,
      'dob': dob,
    };
  }
}
